handlertest/handlertest.go

```diff
@@ -0,0 +1,161 @@
+package handlertest
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	toolkit "github.com/fastbill/go-service-toolkit"
+	"github.com/fastbill/go-service-toolkit/observance"
+	"github.com/labstack/echo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+// Suite holds the general properties to run handler tests.
+type Suite struct {
+	DefaultMiddleware []echo.MiddlewareFunc
+	DefaultHeaders    map[string]string
+}
+
+// Params contains all settings that should be used when calling the handler.
+type Params struct {
+	Route      string
+	Method     string
+	Body       interface{}
+	Headers    map[string]string
+	PathParams []PathParam
+	Query      map[string]string
+	Middleware []echo.MiddlewareFunc
+}
+
+// MockAsserter is the interface that needs to be fulfilled by the mocks that are passed in.
+type MockAsserter interface {
+	AssertExpectations(t mock.TestingT) bool
+}
+
+// PathParam defines the name and value for a parameter in the URL.
+type PathParam struct {
+	Name  string
+	Value string
+}
+
+// CallHandler calls the given handler with the provided settings.
+// Afterwards it calls "AssertExpectations" of the provided mocks.
+func (s *Suite) CallHandler(t *testing.T, handlerFunc echo.HandlerFunc, params *Params, mocks []MockAsserter) (*httptest.ResponseRecorder, error) {
+	if handlerFunc == nil {
+		return nil, errors.New("error in test setup, handler missing")
+	}
+
+	if params == nil {
+		params = &Params{}
+	}
+	route, method := defineRouteAndMethod(params, t)
+
+	bodyAsReader, err := convertToReader(params.Body)
+	assert.NoError(t, err, "error in test setup in CallHandler")
```

Comment:
We could go here with `require.NoError`.

Reply:
I tried but it does not work well. If the assertion fails it creates a panic without any information what happened. Not sure why. You can try it yourself, change it and run the tests.

I did not feel it is worth to invest more time into the issue at this point.